### PR TITLE
DDF-2444 Added anyTags filter to include results with empty tags

### PR DESCRIPTION
**Pull request by emanns95**
#### What does this PR do?
Fixes itest failures caused by filters on Metacard.TAGS removing metacards without tags from the query.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard 
@mcalcote 
@spearskw 
@rzwiefel 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@figliold
@lessarderic

#### How should this be tested?
Full build and tests

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by emanns95** [comment]
Not ready to merge yet

**Comment by coyotesqrl** [comment]
Does the fix, when it is vetted and complete, need to be backported to 2.9?

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java [hunk 1]
```diff
@@ -665,11 +663,24 @@ protected ResourceInfo getResourceInfo(ResourceRequest resourceRequest, String s
...
+
+        //no metacard tag
+        Filter nullTag = frameworkProperties.getFilterBuilder()
+                .attribute(Metacard.TAGS)
+                .is()
+                .empty();
+
+        //any or no metacard tag
+        anyTag = frameworkProperties.getFilterBuilder()
+                .anyOf(anyTag, nullTag);
```
Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java]: Just a bit, don't overwrite the variable. Makes it harder to follow. Declare a new var for final filter.

**Comment by rzwiefel** [comment]
Also this is a pr for catalog framework, touching a specific persons code. You should be adding the person who's code it was originally as a reviewer. EG, me.  Also since it's catalog framework you should add the respective component owner as a committer (EG, Phil. Possibly Richard but not sure he feels comfortable with that burden yet)

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java [hunk 2]
```diff
@@ -665,11 +663,24 @@ protected ResourceInfo getResourceInfo(ResourceRequest resourceRequest, String s
     }
 
     private Query anyTag(Query query) {
+        //Any metacard tag
```
Comment (clockard) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java]: Need to do some kind of check to to see if this is a local query. If it is not then the tag filters shouldn't be applied.

**Comment by emanns95** [comment]
Regarding backporting to 2.9, the change this is modifying is not on that branch, so I'd say no need.

**Comment by emanns95** [comment]
@rzwiefel is going to be coming up with a different fix for this

**Pull request closed without merging**